pkg/reader/ccip_interface.go

```diff
@@ -133,7 +133,7 @@ type CCIPReader interface {
 	ExecutedMessages(
 		ctx context.Context,
 		source cciptypes.ChainSelector,
```

Comment:
I think the func signature should be the following:

```suggestion
		ranges map[cciptypes.ChainSelector]SeqNumRange,
```

This allows us to query for ExecutionStateChanged messages emitted for potentially different chains in the same optimized batch query.